internal/pkg/queue/enqueue.go

```diff
@@ -0,0 +1,133 @@
+package queue
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"sync"
+)
+
+// BatchEnqueue adds 1 or many items to the queue in a single operation.
+// If multiple items are provided, the order in which they will be enqueued is not guaranteed.
+// It WILL be less efficient than Enqueue for single items.
+func (q *PersistentGroupedQueue) BatchEnqueue(items ...*Item) error {
+	if items == nil {
+		return errors.New("cannot enqueue nil item")
+	}
+
+	if !q.CanEnqueue() {
+		return ErrQueueClosed
+	}
+
+	q.mutex.Lock()
+	defer q.mutex.Unlock()
+
+	// Seek to end of file to get start position
+	startPos, err := q.queueFile.Seek(0, io.SeekEnd)
+	if err != nil {
+		return fmt.Errorf("failed to seek to end of file: %s", err.Error())
+	}
+
+	// Encode all items in parallel
+	type msg struct {
+		bytes []byte
+		item  *Item
+	}
+
+	itemsChan := make(chan *msg, len(items))
+	wg := &sync.WaitGroup{}
+	wg.Add(len(items))
+
+	for _, item := range items {
+		if item == nil {
+			q.logger.Error("cannot enqueue nil item")
+			continue
+		}
+
+		go func(i *Item, wg *sync.WaitGroup) {
+			defer wg.Done()
+
+			b, err := encodeItem(i)
+			if err != nil {
+				q.logger.Error("failed to encode item", "err", err)
+				return
+			}
+
+			itemsChan <- &msg{
+				bytes: b,
+				item:  i,
+			}
+		}(item, wg)
+	}
+
+	wg.Wait()
+	close(itemsChan)
```

Comment:
@CorentinB what in the banana ham jelly bagel sandwich is that, you close a channel BEFORE reading it? and you wonder why we don't crawl much

Reply:
Question is: how does it even work right now

Reply:
@equals215 it's perfectly fine to do that, the channel is closed but the elements are still in it and can be consumed, the range just after is working fine